identity/src/main/java/com/android/identity/L2CAPClient.java

```diff
@@ -147,41 +140,54 @@ private void writeToSocket() {
                 OutputStream os = mSocket.getOutputStream();
                 os.write(messageToSend);
                 os.flush();
-
+                reportMessageSendProgress(messageToSend.length, messageToSend.length);
+                mLog.transport(String.format(Locale.US, "Wrote CBOR data item of size %d bytes",
+                        messageToSend.length));
             } catch (IOException e) {
                 Log.e(TAG, "Error writing message on L2CAP socket", e);
                 reportError(new Error("Error writing message on L2CAP socket", e));
                 return;
             }
+            // TODO: If we don't do this we run into problems when sending a session termination
```

Comment:
`BluetoothSocket.java` is just a thin Java layer communicating with native code using binder.

Reply:
I was thinking there was maybe a race with closing the socket or something, but that seems impossible. We join this thread.

We should also check with the BT team on this, because we should never have to sleep after a flush.